Products/PasswordStrength/plugin.py

```diff
@@ -1,280 +1,242 @@
+                                     'PasswordStrength+plugin+added.'
+                                     % dispatcher.absolute_url())
+
+DEFAULT_POLICIES = [(r'.{10}.*', 'Minimum 10 characters'),
+                    (r'.*[A-Z].*', 'Minimum 1 capital letter'),
+                    (r'.*[a-z].*', 'Minimum 1 lower case letter'),
+                    (r'.*[0-9].*', 'Minimum 1 number'),
+                    (r'.*[^0-9a-zA-Z ].*', 'Minimum 1 non-alpha character'),
+                    ]
+
+
+class PasswordStrength(BasePlugin, Cacheable):
+
+    """PAS plugin that ensures strong passwords
+    """
+
+    meta_type = 'Password Strength Plugin'
+    security = ClassSecurityInfo()
+
+    _properties = ({'id': 'title',
+                    'label': 'Title',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p1_re',
+                    'label': 'Policy 1 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p1_err',
+                    'label': 'Policy 1 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p2_re',
+                    'label': 'Policy 2 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p2_err',
+                    'label': 'Policy 2 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p3_re',
+                    'label': 'Policy 3 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p3_err',
+                    'label': 'Policy 3 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p4_re',
+                    'label': 'Policy 4 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p4_err',
+                    'label': 'Policy 4 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p5_re',
+                    'label': 'Policy 5 Regular Expression',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   {'id': 'p5_err',
+                    'label': 'Policy 5 Error Message',
+                    'type': 'string',
+                    'mode': 'w',
+                    },
+                   )
+
+    def __init__(self, id, title=None):
+        self._id = self.id = id
+        self.title = title
+
+        i = 1
+        for reg, err in DEFAULT_POLICIES:
```

Comment:
`for i, (reg , err) in enumerate(DEFAULT_POLICIES, 1):`

You can use enumerate to factor out your manual index handling here